### DISCUSSION TITLE
文件类型如何与Scheme关联？

### DISCUSSION THREAD

**Discussion by lusida** [opensumi/core#1751] [Q&A] [answered]
https://opensumi.com/zh/docs/develop/module-apis/editor 这个案例中调用open(URI)可以打开自定义schame对应的Component，那什么让Explorer中的文件双击打开自动关联对应的schame呢？比如文件后缀 .xx对应的schame-xx什么配置？

**Comment by erha19** [comment, marked as answer]
@lusida 参考一下这里案例的写法，刚刚写了一个例子 https://github.com/opensumi/opensumi-module-samples/pull/11

**Reply by erha19** [reply]
这里核心思路即是通过对 `Schemes.File` 协议文件的打开追加一个内容识别逻辑，从而实现打开自定义组件的能力，关键代码：
```ts
// Handle `xxx.ant-theme` files with custom component.
registry.registerEditorComponentResolver(Schemes.file, (resource, results) => {
  if (resource.uri.path.ext === `.${COMPONENTS_SCHEME_ID}`) {
    results.push({
      type: 'component',
      componentId: COMPONENTS_ID,
    });
  }
});
```
